### DISCUSSION TITLE
Connecting to on-premise SharePoint Subscription edition

### DISCUSSION THREAD

**Discussion by StretchFredrik** [pnp/pnpcore#1085] [Q&A] [answered]
Hello!

Is there any way to connect to an on-prem subscription edition SharePoint using this sdk? If so, how would one go about it?

Thank you for your time! :)

**Comment by jansenbe** [comment, marked as answer]
@StretchFredrik : this is not possible, we only support SharePoint Online. Authentication requires Azure AD, which is not available for on-premises. Furthermore we use a lot of Graph APIs and they also are not available for on-premises.

**Reply by StretchFredrik** [reply]
Are there any alternatives for on-premise that you know of? Everything i find is deprecated.

**Reply by jansenbe** [reply]
The old PnP Core library for SharePoint 2019 is still there (https://www.nuget.org/packages/SharePointPnPCore2019). We've stopped to maintain this one 2 years ago, but it should still work.

**Reply by Alansoft-BR** [reply]
So there is absolutely nothing for .NET Standard that works for on-premise SP? This is driving me crazy.
